File: web/src/api/client/diaryTag.js
import service from '@/utils/request'

export const createDiaryTag = (data) => {
  return service({
    url: '/diaryTag/createDiaryTag',
    method: 'post',
    data
  })
}

export const deleteDiaryTag = (data) => {
  return service({
    url: '/diaryTag/deleteDiaryTag',
    method: 'delete',
    data
  })
}

export const deleteDiaryTagByIds = (data) => {
  return service({
    url: '/diaryTag/deleteDiaryTagByIds',
    method: 'delete',
    data
  })
}

export const updateDiaryTag = (data) => {
  return service({
    url: '/diaryTag/updateDiaryTag',
    method: 'put',
    data
  })
}

export const findDiaryTag = (params) => {
  return service({
    url: '/diaryTag/findDiaryTag',
    method: 'get',
    params
  })
}

export const getDiaryTagList = (params) => {
  return service({
    url: '/diaryTag/getDiaryTagList',
    method: 'get',
    params
  })
}

export const getDiaryTagPublic = (params) => {
  return service({
    url: '/diaryTag/getDiaryTagPublic',
    method: 'get',
    params
  })
}

File: server/router/client/diary_tag.go
package client

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type DiaryTagRouter struct{}

// InitDiaryTagRouter 初始化 日记标签 路由信息
func (r *DiaryTagRouter) InitDiaryTagRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	diaryTagRouter := Router.Group("diaryTag").Use(middleware.OperationRecord())
	diaryTagRouterWithoutRecord := Router.Group("diaryTag")
	diaryTagRouterWithoutAuth := PublicRouter.Group("diaryTag")
	{
		diaryTagRouter.POST("createDiaryTag", diaryTagApi.CreateDiaryTag)
		diaryTagRouter.DELETE("deleteDiaryTag", diaryTagApi.DeleteDiaryTag)
		diaryTagRouter.DELETE("deleteDiaryTagByIds", diaryTagApi.DeleteDiaryTagByIds)
		diaryTagRouter.PUT("updateDiaryTag", diaryTagApi.UpdateDiaryTag)
	}
	{
		diaryTagRouterWithoutRecord.GET("findDiaryTag", diaryTagApi.FindDiaryTag)
		diaryTagRouterWithoutRecord.GET("getDiaryTagList", diaryTagApi.GetDiaryTagList)
	}
	{
		diaryTagRouterWithoutAuth.GET("getDiaryTagPublic", diaryTagApi.GetDiaryTagPublic)
	}
}